src/Nethermind/Nethermind.Consensus/Transactions/InclusionListTxSource.cs

```diff
@@ -0,0 +1,27 @@
+// SPDX-FileCopyrightText: 2025 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Nethermind.Consensus.Producers;
+using Nethermind.Core;
+using Nethermind.Crypto;
+using Nethermind.Serialization.Rlp;
+
+namespace Nethermind.Consensus.Transactions;
+
+public class InclusionListTxSource(ulong chainId) : ITxSource
+{
+    private readonly EthereumEcdsa _ecdsa = new(chainId);
+
+    public IEnumerable<Transaction> GetTransactions(BlockHeader parent, long gasLimit, PayloadAttributes? payloadAttributes = null)
+        => payloadAttributes?.InclusionListTransactions?.Select(tx => DecodeTransaction(tx)) ?? [];
```

Comment:
We decode it in `ComputePayloadId` and we decode it here again, looks like double the work?

I think we could decode in parallel? It should be faster, especially with address recovery.